py3status/modules/uptime.py

```diff
@@ -63,63 +63,33 @@ class Py3status:
-        self._decades = self.py3.format_contains(self.format, 'decades')
-        self._years = self.py3.format_contains(self.format, 'years')
-        self._weeks = self.py3.format_contains(self.format, 'weeks')
-        self._days = self.py3.format_contains(self.format, 'days')
-        self._hours = self.py3.format_contains(self.format, 'hours')
-        self._minutes = self.py3.format_contains(self.format, 'minutes')
-        self._seconds = self.py3.format_contains(self.format, 'seconds')
+        self.time_periods = OrderedDict()
+        periods = [
+            ('decades', 315360000),
+            ('years', 31536000),
+            ('weeks', 604800),
+            ('days', 86400),
+            ('hours', 3600),
+            ('minutes', 60),
+            ('seconds', 1),
+        ]
+        for unit, second in periods:
+            if self.py3.format_contains(self.format, unit):
+                self.time_periods[unit] = second
 
     def uptime(self):
-        # Units will be computed from bare seconds since timedelta only
-        # provides .days and .seconds anyway. Getting rid of the seconds
-        # part. Keeping the floating point part would make divmod return
-        # floats, and thus would require days/hours/minutes/seconds to be
-        # casted to int before formatting, which would be dirty to handle
-        # since we can't cast None to int.
         with open('/proc/uptime', 'r') as f:
             up = int(float(f.readline().split()[0]))
             offset = time() - up
 
-        cache_timeout = decades = years = weeks = days = hours = minutes = seconds = 0
-
-        # Decades
-        if self._decades:
-            decades, up = divmod(up, 315360000)  # 10 years -> decade
-            cache_timeout = 315360000
-        # Years
-        if self._years:
-            years, up = divmod(up, 31536000)  # 365 days -> year
-            cache_timeout = 31536000
-        # Weeks
-        if self._weeks:
-            weeks, up = divmod(up, 604800)  # 7 days -> week
-            cache_timeout = 604800
-        # Days
-        if self._days:
-            days, up = divmod(up, 86400)  # 24 hours -> day
-            cache_timeout = 86400
-        # Hours
-        if self._hours:
-            hours, up = divmod(up, 3600)  # 60 minutes -> hour
-            cache_timeout = 3600
-        # Minutes
-        if self._minutes:
-            minutes, up = divmod(up, 60)  # 60 seconds -> minute
-            cache_timeout = 60
-        # Seconds
-        if self._seconds:
-            seconds = up  # 1000000000 nanoseconds -> second
-            cache_timeout = 1
-
-        uptime = self.py3.safe_format(self.format, dict(
-            decades=decades, years=years, weeks=weeks, days=days,
-            hours=hours, minutes=minutes, seconds=seconds))
+        uptime = {}
+        for unit in self.time_periods:
+            uptime[unit], up = divmod(up, self.time_periods[unit])
+            cache_to = self.time_periods[unit]
 
```

Comment:
```
cache_to = self.time_periods[unit]
uptime[unit], up = divmod(up, cache_to)
```

is cleaner maybe we should call it `time_interval` as that has more meaning

Reply:
Reached end... Right on `79` with `cache_to`... and a little cute wordplay.. `sync_to... cache_to`. 
```
        return {
            'cached_until': self.py3.time_in(sync_to=cache_to, offset=offset),
            'full_text': self.py3.safe_format(self.format, uptime)
        }
```

Reply:
cute has no friends in coding...

meaningful names win every time

Reply:
`interval` fits in the column too. Pushing...